### ISSUE TITLE
NullPointerException after Eastwood upgrade from 0.6.0 to 0.7.0 

### ISSUE THREAD

**Issue by scarytom** [jonase/eastwood#404]
I just upgraded from 0.6.0 to 0.7.0 and now I'm getting an NPE to do with deprecations, so perhaps related to https://github.com/jonase/eastwood/commit/f430256625f23804cc1ba3364f5ed2d6ba28198d

With Eastwood 0.6.0 I get the following output
```
== Eastwood 0.6.0 Clojure 1.10.3 JVM 11.0.11 ==
Directories scanned for source files:
 
src dev test
== Linting xxx ==
src/xxx.clj:78:19: deprecations: Constructor 'public com.lmax.disruptor.dsl.Disruptor(com.lmax.disruptor.EventFactory,int,java.util.concurrent.Executor,com.lmax.disruptor.dsl.ProducerType,com.lmax.disruptor.WaitStrategy)' is deprecated.
== Linting done in 7922 ms ==
== Warnings: 1 (not including reflection warnings)  Exceptions thrown: 0
Subprocess failed (exit code: 1)
```

With Eastwood 0.7.0 I get this:
```
== Eastwood 0.7.0 Clojure 1.10.3 JVM 11.0.11 ==
Directories scanned for source files:
 
src dev test
== Linting xxx ==
src/xxx.clj:78:19: deprecations: Constructor 'public com.lmax.disruptor.dsl.Disruptor(com.lmax.disruptor.EventFactory,int,java.util.concurrent.Executor,com.lmax.disruptor.dsl.ProducerType,com.lmax.disruptor.WaitStrategy)' is deprecated.
NullPointerException 
	eastwood.passes/get-method (passes.clj:48)
	eastwood.passes/get-method (passes.clj:42)
	eastwood.linters.unused/unused-ret-val-lint-result (unused.clj:417)
	eastwood.linters.unused/unused-ret-val-lint-result (unused.clj:391)
	eastwood.linters.unused/unused-ret-vals-2/iter--10516--10520/fn--10521/fn--10522 (unused.clj:559)
	eastwood.linters.unused/unused-ret-vals-2/iter--10516--10520/fn--10521 (unused.clj:511)
	clojure.lang.LazySeq.sval (LazySeq.java:42)
	clojure.lang.LazySeq.seq (LazySeq.java:51)
	clojure.lang.RT.seq (RT.java:535)
	clojure.core/seq--5419 (core.clj:139)
	clojure.core/filter/fn--5911 (core.clj:2813)
	clojure.lang.LazySeq.sval (LazySeq.java:42)
	clojure.lang.LazySeq.seq (LazySeq.java:58)
	clojure.lang.RT.seq (RT.java:535)
	clojure.core/seq--5419 (core.clj:139)
	clojure.core/seq--5419 (core.clj:139)
	eastwood.linters.unused/unused-ret-vals*/iter--10559--10563/fn--10564 (unused.clj:585)
	clojure.lang.LazySeq.sval (LazySeq.java:42)
	clojure.lang.LazySeq.seq (LazySeq.java:51)
	clojure.lang.RT.seq (RT.java:535)
	clojure.core/seq--5419 (core.clj:139)
	clojure.core/map/fn--5884 (core.clj:2750)
	clojure.lang.LazySeq.sval (LazySeq.java:42)
	clojure.lang.LazySeq.seq (LazySeq.java:51)
	clojure.lang.RT.seq (RT.java:535)
	clojure.core/seq--5419 (core.clj:139)
	clojure.core/filter/fn--5911 (core.clj:2813)
	clojure.lang.LazySeq.sval (LazySeq.java:42)
	clojure.lang.LazySeq.seq (LazySeq.java:51)
	clojure.lang.RT.seq (RT.java:535)
	clojure.core/seq--5419 (core.clj:139)
	clojure.core/dorun (core.clj:3121)
	clojure.core/doall (core.clj:3136)
	clojure.core/doall (core.clj:3136)
	eastwood.lint/run-linter (lint.clj:164)
	eastwood.lint/run-linter (lint.clj:161)
	eastwood.lint/lint-ns* (lint.clj:174)
	eastwood.lint/lint-ns* (lint.clj:173)
	clojure.core/partial/fn--5861 (core.clj:2642)
	clojure.core/pmap/fn--8485/fn--8486 (core.clj:7024)
	clojure.core/binding-conveyor-fn/fn--5772 (core.clj:2034)
	java.util.concurrent.FutureTask.run (FutureTask.java:264)
	java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1128)
	java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:628)
	java.lang.Thread.run (Thread.java:829)

NullPointerException 
	eastwood.passes/get-method (passes.clj:48)
	eastwood.passes/get-method (passes.clj:42)
	eastwood.linters.unused/unused-ret-val-lint-result (unused.clj:417)
	eastwood.linters.unused/unused-ret-val-lint-result (unused.clj:391)
	eastwood.linters.unused/unused-ret-vals-2/iter--10516--10520/fn--10521/fn--10522 (unused.clj:559)
	eastwood.linters.unused/unused-ret-vals-2/iter--10516--10520/fn--10521 (unused.clj:511)
	clojure.lang.LazySeq.sval (LazySeq.java:42)
	clojure.lang.LazySeq.seq (LazySeq.java:51)
	clojure.lang.RT.seq (RT.java:535)
	clojure.core/seq--5419 (core.clj:139)
	clojure.core/filter/fn--5911 (core.clj:2813)
	clojure.lang.LazySeq.sval (LazySeq.java:42)
	clojure.lang.LazySeq.seq (LazySeq.java:58)
	clojure.lang.RT.seq (RT.java:535)
	clojure.core/seq--5419 (core.clj:139)
	clojure.core/seq--5419 (core.clj:139)
	eastwood.linters.unused/unused-ret-vals*/iter--10559--10563/fn--10564 (unused.clj:585)
	clojure.lang.LazySeq.sval (LazySeq.java:42)
	clojure.lang.LazySeq.seq (LazySeq.java:51)
	clojure.lang.RT.seq (RT.java:535)
	clojure.core/seq--5419 (core.clj:139)
	clojure.core/map/fn--5884 (core.clj:2750)
	clojure.lang.LazySeq.sval (LazySeq.java:42)
	clojure.lang.LazySeq.seq (LazySeq.java:51)
	clojure.lang.RT.seq (RT.java:535)
	clojure.core/seq--5419 (core.clj:139)
	clojure.core/filter/fn--5911 (core.clj:2813)
	clojure.lang.LazySeq.sval (LazySeq.java:42)
	clojure.lang.LazySeq.seq (LazySeq.java:51)
	clojure.lang.RT.seq (RT.java:535)
	clojure.core/seq--5419 (core.clj:139)
	clojure.core/dorun (core.clj:3121)
	clojure.core/doall (core.clj:3136)
	clojure.core/doall (core.clj:3136)
	eastwood.lint/run-linter (lint.clj:164)
	eastwood.lint/run-linter (lint.clj:161)
	eastwood.lint/lint-ns* (lint.clj:174)
	eastwood.lint/lint-ns* (lint.clj:173)
	clojure.core/partial/fn--5861 (core.clj:2642)
	clojure.core/pmap/fn--8485/fn--8486 (core.clj:7024)
	clojure.core/binding-conveyor-fn/fn--5772 (core.clj:2034)
	java.util.concurrent.FutureTask.run (FutureTask.java:264)
	java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1128)
	java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:628)
	java.lang.Thread.run (Thread.java:829)

== Linting done in 7435 ms ==
== Warnings: 1 (not including reflection warnings)  Exceptions thrown: 2
Subprocess failed (exit code: 1)

```

Furthermore, with 0.6.0 I'm able to suppress the failure using the following config:
```(disable-warning
  {:linter :deprecations
   :symbol-matches #{#".*com\.lmax\.disruptor\.dsl\.Disruptor.*"}
   :reason "We use the constructor that takes an Executor so that we can use a cached thread pool."})
```
This suppression does not work with the 0.7.0 NPE

**Comment by vemv** [comment]
Thanks so much for the report. It's encouraging to see people upgrading so fast 🍻

Will fix+release this in a bit

**Comment by vemv** [comment]
0.7.1 has been released. It contains what (I hope) is a fix for the NPE. 

I'd say the NPE's cause is related to https://github.com/jonase/eastwood/commit/108978a151ced250dc9bc8e7154664ab79aef02f . That commit introduces a new lintable thing (namely instance method calls), so it's plausible that they can fail for some edge cases.

(we try for that not to happen by having more exhaustive CI by the day)

As for :symbol-matches, it keeps working and seemingly it didn't need a fix. I added a new unit test just to make sure.

If that's not the case, I'd appreciate a new issue specifying the Maven dependency I should download for `Disruptor`.
